### ISSUE TITLE
ssh2.sh test failed with OpenSSH 8.8p1

### ISSUE THREAD

**Issue by icebluey** [libssh2/libssh2#630]
libssh2 version: 1.9.0 and 1.10.0
```
[root@localhost]# make check
...
PASS: simple
PASS: mansyntax.sh
FAIL: ssh2.sh
============================================================================
Testsuite summary for libssh2 -
============================================================================
# TOTAL: 3
# PASS:  2
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See tests/test-suite.log
Please report to libssh2-devel@cool.haxx.se
============================================================================



[root@localhost]# cat tests/test-suite.log
=====================================
   libssh2 -: tests/test-suite.log
=====================================

# TOTAL: 3
# PASS:  2
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: ssh2.sh
=============

command-line line 0: Deprecated option UsePrivilegeSeparation
command-line line 0: Deprecated option UsePrivilegeSeparation
Failure establishing SSH session
FAIL ssh2.sh (exit status: 1)

[root@localhost]# ssh -V
OpenSSH_8.8p1, OpenSSL 1.1.1l  24 Aug 2021

```

**Comment by willco007** [comment]
Could you add the error output to this report or the ssh error logs?

**Comment by kanavin** [comment]
ssh2.sh is the failing test in all of them, I'm just running it manually in a standalone target install. And the issue seems similar too:

Unable to negotiate with 127.0.0.1 port 57322: no matching host key type found. Their offer: ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ecdsa-sha2-nistp256-cert-v01@openssh.com,ecdsa-sha2-nistp]

**Comment by willco007** [comment]
Rebuilding from the latest changes in https://github.com/libssh2/libssh2/pull/626 _should_ resolve this issue.

**Comment by Zenju** [comment]
> Rebuilding from the latest changes in #626 _should_ resolve this issue.

For what it's worth, the latest official release libssh 1.10.0 is broken for OpenSSH_8.8p1 servers and newer (e.g. on openSUSE Tumbleweed) when trying to log in via key file (libssh2_userauth_publickey_frommemory), instead of user/password.
However these changes seem to fix the issue when testing today's master branch!

**Comment by willco007** [comment]
@Zenju Thanks for the confirmation!  I'm going to close this as fixed with #626.
